datamarket/components/MyDatasets/DatasetActions.tsx

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { DatasetWithIndex } from "../../constants";
+import DownloadIcon from "@mui/icons-material/Download";
+import VisibilityIcon from "@mui/icons-material/Visibility";
+import EditIcon from "@mui/icons-material/Edit";
+import {
+  decryptAndDownload,
+} from "../../utils/utils";
+import { Button } from "../Button";
+import { useRouter } from "next/router";
+
+export const DatasetActions = ({
+  isOwner,
+  publicIndex,
+  dataset,
+  accounts,
+  showIcons,
+}: {
+    isOwner?: boolean;
+    publicIndex: number | undefined;
+    dataset: DatasetWithIndex;
+    accounts: string[];
+    showIcons: boolean;
+}) => {
+  const router = useRouter();
+  const navToView = (): void => {
+    router.push(`view/${dataset.sample}`);
+  };
+  const navToUpdate = (): void => {
+    router.push(`update/${publicIndex}`);
+  };
+
+  const downloadDataset = (): void => {
+    decryptAndDownload({ cid: dataset.data, account: accounts[0] });
+  };
+  return (
+    <div className="">
+      {showIcons && (
```

Comment:
if we dont want to render anything when showIcons is false, then it should be
```
if (!showIcons) {
return <></>
}
return (
the actual thing
)
```

Reply:
Done